### PR TITLE
[SPARK-43252][SQL] Rename the error class _LEGACY_ERROR_TEMP_2017 to UNRESOLVED_CUSTOM_CLASS

### PR DESCRIPTION
**Pull request by luizfna**
What changes were proposed in this pull request?
In the PR, I propose to assign the proper name UNRESOLVED_CUSTOM_CLASS to the legacy error class _LEGACY_ERROR_TEMP_2017 , and modify test suite to use checkError() which checks the error class name.

Why are the changes needed?
Proper name improves user experience with Spark.

Does this PR introduce any user-facing change?
Yes, the PR changes an user-facing error message.

How was this patch tested?
By running the modified test suites:

$ build/sbt "testOnly *ObjectExpressionsSuite"

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1599,6 +1599,11 @@
     },
     "sqlState" : "42703"
   },
+  "UNRESOLVED_CUSTOM_CLASS": {
```
Comment (Hisoka-X) [inline core/src/main/resources/error/error-classes.json:1602]: Is it better to put the class name that cannot be parsed in the message?

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -1599,6 +1599,11 @@
...
+    "message": [
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1603]:
```suggestion
    "message" : [
```

Comment (MaxGekk) [reply core/src/main/resources/error/error-classes.json:1603]: Could you run:
```
$ build/sbt "core/testOnly *SparkThrowableSuite"
```
It should catch the format errors in `error-classes.json`

File: core/src/main/resources/error/error-classes.json [hunk 3]
```diff
@@ -1599,6 +1599,11 @@
...
+      "Could not resolve custom collection class. Please ensure that the class name is correct and that the class is available on the classpath."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1604]:
```suggestion
      "Cannot resolve the custom collection class: <className>. Please ensure that the class name is correct and it is available on the classpath."
```

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala [hunk 4]
```diff
@@ -459,6 +459,19 @@ class ObjectExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
...
   }
 
+  test("UnresolvedMapObjects should support interpreted execution") {
+    val inputExpr = Literal(1)
+    val func = (expr: Expression) => expr
+    val customCls: Option[Class[_]] = None
+
+    checkError(
+      exception = intercept[SparkUnsupportedOperationException] {
+        UnresolvedMapObjects(func, inputExpr, customCls).dataType
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala:469]: Could you try to trigger the error from user code. BTW, the error classes should be assigned to user-facing errors.

**Review by MaxGekk — CHANGES_REQUESTED** [review]
@luizfna Could you restore the formatting of PR description, and the `[SQL]` tag to PR's title.

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (Hisoka-X) [reply core/src/main/resources/error/error-classes.json:1602]: I think there two different case, 1. class name is null 2. class name isn't null, but can't be resolved. The `UNRESOLVED_CUSTOM_CLASS` only suite for case 2. Case 1 should be handle before resolve class name. @MaxGekk WDYT?

Comment (luizfna) [reply core/src/main/resources/error/error-classes.json:1602]: In my opinion, it seems impossible to cause the error to occur unless the class is null. Also when I try I non existing class or a class not defined in the corresponding PATH I get: ClassNotFoundException. What do you guys think about letting the error without the class and maybe adjusting the message?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala (hunk 4 above)
Comment (luizfna) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala:469]: I couldn't do something interesting in that matters. Should I proceed with SparkException.internalError() approach?

**Comment by MaxGekk** [comment]
@luizfna Could you address comments, please.

**Comment by luizfna** [comment]
@MaxGekk I would like some calcifications regarding the discussions before to do it so

**Comment by dengziming** [comment]
The title should be SPARK-43253 instead of SPARK-43252.

**Pull request closed without merging**